deepbox/src/main/java/ch/cyberduck/core/deepbox/DeepboxIdProvider.java

```diff
@@ -82,14 +93,53 @@ public String getBoxNodeId(final Path file) throws BackgroundException {
         return this.getFileId(box);
     }
 
-    public String getThirdLevelId(final Path file) throws BackgroundException {
+    public String getFourthLevelId(final Path file) throws BackgroundException {
         final Path path = containerService.getFourthLevelPath(file);
         if(null == path) {
             throw new NotfoundException(file.getName());
         }
         return this.getFileId(path);
     }
 
+    protected Path normalize(final Path file) {
+        if(!containerService.isInSharedWithMe(file)) {
+            return file;
+        }
+        if(containerService.isSharedWithMe(file)) {
+            return file;
+        }
+        final Deque<Path> segments = this.decompose(file);
+        Path result = new Path(String.valueOf(Path.DELIMITER), EnumSet.of(Path.Type.volume, Path.Type.directory));
+
+        while(!segments.isEmpty()) {
+            Path segment = segments.pop();
+            if(containerService.isSharedWithMe(segment)) {
+                final String combined = segments.pop().getName();
+                final Matcher matcher = SHARED.matcher(combined);
+                if(matcher.matches()) {
+                    final String deepboxName = matcher.group(1);
+                    final String boxName = matcher.group(2);
+                    final EnumSet<AbstractPath.Type> type = EnumSet.copyOf(segment.getType());
+                    type.add(AbstractPath.Type.shared);
+                    final Path deepbox = new Path(result, deepboxName, type, new PathAttributes(segment.attributes()).withFileId(null));
+                    result = new Path(deepbox, boxName, type, segment.attributes());
+                }
+                else {
+                    log.warn("Could not match pattern");
```

Comment:
Improve logging to include name that did not match pattern.

Reply:
In [212d0c7](https://github.com/iterate-ch/cyberduck/pull/16420/commits/212d0c7693cfd27a369d4a980059b27e2da79ab3).